compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SymbolResolver.java

```diff
@@ -1977,6 +1985,28 @@ private BSymbol resolveOperator(ScopeEntry entry, List<BType> types) {
             if (types.size() == opType.paramTypes.size()) {
                 boolean match = true;
                 for (int i = 0; i < types.size(); i++) {
+                    if ((types.get(i).getKind() == TypeKind.UNION) &&
+                            (opType.paramTypes.get(i).getKind() == TypeKind.UNION)) {
+                        if (types.get(i).isNullable() && opType.paramTypes.get(i).isNullable()) {
+                            Iterator<BType> nilLiftTypes = ((BUnionType) types.get(i)).getMemberTypes().iterator();
+                            BType nilLiftTypeOne = null;
+                            while (nilLiftTypes.hasNext()) {
+                                BType nilLiftType = nilLiftTypes.next();
+                                if (nilLiftType.tag != TypeTags.NIL) {
+                                    nilLiftTypeOne = nilLiftType;
```

Comment:
This logic would use the last type in the union as nil lifted. Order should not matter. Check in types.validNumericTypeExists() as well

```ballerina
  int?|string foo = 1;
  int?|string bar = 2;
  var res = (foo + bar);
```  
works. below fails

```ballerina
  int?|string foo = 1;
  string|int? bar = 2;
  var res = (foo + bar);
```

Reply:
Logic is changed now. Now from here it will only resolve type definitions like this `type?` or `type|()` or `()|type`

Reply:
Anyway this should not be supported right?
```
int?|string foo = 1;
string|int? bar = 2;
var res = (foo + bar);
```